pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/invertedindex/TextIndexHandler.java

```diff
@@ -138,6 +138,7 @@ public void updateIndices(SegmentDirectory.Writer segmentWriter, IndexCreatorPro
     }
     for (String column : columnsToAddIdx) {
       ColumnMetadata columnMetadata = _segmentMetadata.getColumnMetadataFor(column);
+      LOGGER.info("Creating text index from segment: {}, column: {}", segmentName, column);
```

Comment:
Similar to `shouldCreateTextIndex` in creation path, shouldn't there be `shouldDeleteTextIndex` in deletion path?

Reply:
When we disable the text_index for column in configuration, we would like to consider deleting index. However, when we add a new column to configuration, we want to build for newer segments. So, for deletion, we want to delete for older segments too, and so we don't need `shouldDeleteTextIndex`.